lib/cli/ui/prompt.rb

```diff
@@ -48,26 +49,29 @@ class << self
         # Free form question when the answer can be empty
         #   CLI::UI::Prompt.ask('What is your opinion on this question?', allow_empty: true)
         #
-        # Question with answers
-        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python))
+        # Multiple choice question with defined handlers
+        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python)) do |handler|
+        #     handler.add_option('rails')  { |selection| puts selection } => outputs "rails" if selected
+        #     handler.add_option('go')     { |selection| puts selection } => outputs "go" if selected
+        #     handler.add_option('ruby')   { |selection| puts selection } => outputs "ruby" if selected
+        #     handler.add_option('python') { |selection| puts selection } => outputs "python" if selected
         #
         #
-        def ask(question, options: nil, default: nil, is_file: nil, allow_empty: true)
-          if (default && !allow_empty) || (options && (default || is_file))
+        def ask(question, default: nil, is_file: nil, allow_empty: true, &define_options)
```

Comment:
I think we would want to keep `options` and allow options _or_ a block to be given. The block can be used for handlers and the options can be used for returning the option. This also makes this change non-breaking.

Reply:
e.g.
```
a.ask('My Question') do |handler|
  handler.add_option('A') { method }
  handler.add_option('B') { method_2 }
end
# Choosing 'A' would return the evaluation of `method`
```
but 
```
a.ask('My Question', options: %w(A B))
# Choosing 'A' would return the option A
```

This keeps a nice UI and UX for the code regardless of the choice you want to make. Sometimes you want the former, sometimes the latter.